tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/scoping/MultipleModulesScopingTest.xtend

```diff
@@ -0,0 +1,168 @@
+/**
+ * Copyright (c) 2020 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.ide.tests.scoping
+
+import org.eclipse.n4js.ide.tests.helper.server.AbstractIdeTest
+import org.junit.Test
+
+/**
+ * Tests for building yarn workspaces, in particular certain corner and special cases
+ * such as "yarn link".
+ */
+class MultipleModulesScopingTest extends AbstractIdeTest {
+
+	@Test
+	def void testTwoN4jsModulesWithSameName() throws Exception {
```

Comment:
Nice tests and works everything as expected!

But you might wanna add these two tests for two special cases:
```xtend
	@Test
	def void testN4jsdAndJsModulesWithSameNameInSameSourceFolderOK() throws Exception {
		testWorkspaceManager.createTestProjectOnDisk(
			#[
				"Client" -> '''
					import {C} from "Module";
					C;
				''',
				"Module.n4jsd" -> '''
					export external const C;
				''',
				"Module.js" -> '''
					export const C = 2;
				'''
			]
		);
		startAndWaitForLspServer();
		assertIssues();
	}

	@Test
	def void testDynamicImportWarning() throws Exception {
		testWorkspaceManager.createTestProjectOnDisk(
			#[
				"Client" -> '''
					import * as M+ from "Module";
					M.C;
				''',
				"Module.n4jsd" -> '''
					export external const C;
				''',
				"Module.js" -> '''
					export const C = 2;
				'''
			]
		);
		startAndWaitForLspServer();
		assertIssues("Client" -> #["(Warning, [0:7 - 0:14], N4JSD module Module should not be imported dynamically.)"]);
	}
```

Reply:
Thanks, I added the tests!